aiohttp/connector.py

```diff
@@ -392,11 +396,18 @@ def closed(self):
 
             try:
                 await fut
-            finally:
-                # remove a waiter even if it was cancelled
-                waiters.remove(fut)
-                if not waiters:
-                    del self._waiters[key]
+            except BaseException:
+                # remove a waiter even if it was cancelled, normally it's
```

Comment:
if the future has been canceled, we do need to wake up another waiter. Take a look at the semaphore implementation [1]

[1] https://github.com/python/cpython/blob/master/Lib/asyncio/locks.py#L478

Reply:
this method created the future, why would we need to wake up another waiter?  That doesn't make sense as it would imply yet another connector is available.  This is 1-1, one waiter was added, one removed.

Reply:
My mistake, the wake up will be done automariaclly by the exit of the context manager in whatever scenario. So forget about this